internal/datacoord/server.go

```diff
@@ -353,39 +354,50 @@ func (s *Server) stopCompactionTrigger() {
 }
 
 func (s *Server) initGarbageCollection() error {
-	var cli *minio.Client
-	var err error
-	if Params.DataCoordCfg.EnableGarbageCollection {
-		var creds *credentials.Credentials
-		if Params.MinioCfg.UseIAM {
-			creds = credentials.NewIAM(Params.MinioCfg.IAMEndpoint)
-		} else {
-			creds = credentials.NewStaticV4(Params.MinioCfg.AccessKeyID, Params.MinioCfg.SecretAccessKey, "")
-		}
-		// TODO: We call minio.New in different places with same procedures to call several functions.
-		// We should abstract this to a focade function to avoid applying changes to only one place.
-		cli, err = minio.New(Params.MinioCfg.Address, &minio.Options{
-			Creds:  creds,
-			Secure: Params.MinioCfg.UseSSL,
-		})
-		if err != nil {
-			return fmt.Errorf("failed to create minio client: %v", err)
+	var cli storage.ChunkManager
+	if Params.CommonCfg.StorageType == "minio" {
```

Comment:
Because there are some special process to initial minio client here, such as retry times of CheckBucketFn().

Reply:
> Because there are some special process to initial minio client here, such as retry times of CheckBucketFn().

maybe extend `ChunkManagerFactory` is a better way

Reply:
ok, after the discussion with yuchen, we use the chunkmanagerfactory